### PR TITLE
add --copy-tests flag for copying installed tests to curr dir.

### PR DESCRIPTION
**Pull request by rwcarlsen**
This is necessary for people with only level-1 access to various moose
apps/codes - the tests will be installed into a location where they
don't have write access - so the test harness can't run them there.
Running this flag copies the installed tests out to an appname_tests dir
that users can cd into and then run them.

ref #16411

### CODE REVIEW

**Review by loganharbour — CHANGES_REQUESTED** [review]
All minor. Looks good to me

**Comment by loganharbour** [comment]
What will happen if someone runs `--copy-tests` on a binary that hasn't been installed?

File: framework/src/base/MooseApp.C [hunk 1]
```diff
@@ -1160,6 +1162,39 @@ MooseApp::run()
...
+                 "\" already exists.\nTo update/recopy tests, rename (\"mv ",
+                 dst_dir,
+                 " new_dir_name\") or remove (\"rm -r ",
+                 dst_dir,
+                 "\") the existing directory.\nThen re-run \"",
+                 cmdname,
+                 " --copy-tests\".");
+
+    std::string cmd = "cp -R " + src_dir + " " + dst_dir;
+    int ret = system(cmd.c_str());
```
[the remark this answers is not in the document]
Comment (rwcarlsen) [reply framework/src/base/MooseApp.C:1190]: windows.

[the remark this answers is not in the document]
Comment (rwcarlsen) [reply framework/src/base/MooseApp.C:1190]: Or anything without `cp`

**Comment by rwcarlsen** [comment]
Good catch - I just pushed up some checks to handle that case more gracefully.

**Comment by loganharbour** [comment]
> Good catch - I just pushed up some checks to handle that case more gracefully.

Shouldn't the error be a little more friendly?

**Comment by rwcarlsen** [comment]
What would you suggest for the message?  There could be multiple reasons that error is hit - I don't want to write a novel about all of them.  Maybe permissions just got messed up somehow, or maybe there isn't an installed binary, or maybe something else went wrong, etc.

**Comment by loganharbour** [comment]
> What would you suggest for the message? There could be multiple reasons that error is hit - I don't want to write a novel about all of them. Maybe permissions just got messed up somehow, or maybe there isn't an installed binary, or maybe something else went wrong, etc.

Fair enough, I think this is valid then.

**Review by loganharbour — APPROVED** [review]

**Pull request merged**